std-bits/table/src/main/java/org/enso/table/data/column/builder/StringBuilder.java

```diff
@@ -1,38 +1,71 @@
 package org.enso.table.data.column.builder;
 
+import org.enso.table.data.column.storage.SpecializedStorage;
 import org.enso.table.data.column.storage.Storage;
 import org.enso.table.data.column.storage.StringStorage;
 import org.enso.table.data.column.storage.type.StorageType;
 import org.enso.table.data.column.storage.type.TextType;
 
 /** A builder for string columns. */
 public class StringBuilder extends TypedBuilderImpl<String> {
+  private final TextType type;
   @Override
   protected String[] newArray(int size) {
     return new String[size];
   }
 
-  public StringBuilder(int size) {
+  public StringBuilder(int size, TextType type) {
     super(size);
+    this.type = type;
   }
 
   @Override
   public StorageType getType() {
-    return TextType.VARIABLE_LENGTH;
+    return type;
   }
 
   @Override
   public void appendNoGrow(Object o) {
-    data[currentSize++] = (String) o;
+    String str = (String) o;
+    if (!type.fits(str)) {
+      str = type.adapt(str);
+
+    }
+
+    data[currentSize++] = str;
   }
 
   @Override
   public boolean accepts(Object o) {
-    return o instanceof String;
+    if (o instanceof String s) {
+      return type.fits(s);
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public void appendBulkStorage(Storage<?> storage) {
+    if (storage.getType() instanceof TextType gotType) {
+      if (type.fitsExactly(gotType)) {
+        if (storage instanceof SpecializedStorage<?>) {
+          // This cast is safe, because storage.getType() == this.getType() == TextType iff storage.T == String
+          @SuppressWarnings("unchecked")
+          SpecializedStorage<String> specializedStorage = (SpecializedStorage<String>) storage;
```

Comment:
Would: 
```suggestion
        if (storage instanceof SpecializedStorage<String> specializedStorage) {
```
not work?

Reply:
No, because of Java type erasure.

```
[IJ]buildStdLibAll
[info] compiling 1 Java source to C:\NBO\enso\std-bits\table\target\scala-2.13\classes ...
[error] C:\NBO\enso\std-bits\table\src\main\java\org\enso\table\data\column\builder\StringBuilder.java:51:1:  error: Storage<CAP#1> cannot be safely cast to SpecializedStorage<String>
[error]         if (storage instanceof SpecializedStorage<String> specializedStorage) {
[error]             ^  where CAP#1 is a fresh type-variable:
[error]     CAP#1 extends Object from capture of ?
[error] 1 error
[warn] frgaal exited with exit code 1
```